server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java

```diff
@@ -40,25 +40,32 @@
  */
 public class NodeInfo extends BaseNodeResponse {
 
-    private Version version;
-    private Build build;
+    private final Version version;
+    private final TransportVersion transportVersion;
+    private final Build build;
 
     @Nullable
-    private Settings settings;
+    private final Settings settings;
 
     /**
      * Do not expose this map to other classes. For type safety, use {@link #getInfo(Class)}
      * to retrieve items from this map and {@link #addInfoIfNonNull(Class, ReportingService.Info)}
      * to retrieve items from it.
      */
-    private Map<Class<? extends ReportingService.Info>, ReportingService.Info> infoMap = new HashMap<>();
+    private final Map<Class<? extends ReportingService.Info>, ReportingService.Info> infoMap = new HashMap<>();
 
     @Nullable
-    private ByteSizeValue totalIndexingBuffer;
+    private final ByteSizeValue totalIndexingBuffer;
 
     public NodeInfo(StreamInput in) throws IOException {
         super(in);
         version = Version.readVersion(in);
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)) {
+            transportVersion = TransportVersion.readVersion(in);
+        } else {
+            assert version.before(Version.V_8_8_0);
```

Comment:
Is it possible to read a >=8.8.0 node info structure from a <8.8.0 ES node?

Reply:
No this should not be possible, and I don't think the assert is necessary